entrypoint.sh

```diff
@@ -1,19 +1,77 @@
-#!/bin/sh
-set -e # Exit immediately if a command exits with a non-zero status
+#!/bin/bash
+set -eu # Increase bash strictness
+set -o pipefail
 
-if [ -n "${GITHUB_WORKSPACE}" ] ; then
+if [[ -n "${GITHUB_WORKSPACE}" ]]; then
   cd "${GITHUB_WORKSPACE}/${INPUT_WORKDIR}" || exit
 fi
 
 export REVIEWDOG_GITHUB_API_TOKEN="${INPUT_GITHUB_TOKEN}"
 
-remark --frail --quiet --use=remark-preset-lint-recommended . ${INPUT_REMARK_FLAGS} 2>&1 |
+# If no arguments are given use current working directory
+remark_args=(".")
+if [[ "$#" -eq 0 && "${INPUT_REMARK_FLAGS}" != "" ]]; then
+  remark_args+=(${INPUT_REMARK_FLAGS})
+elif [[ "$#" -ne 0 && "${INPUT_REMARK_FLAGS}" != "" ]]; then
+  remark_args+=($* ${INPUT_REMARK_FLAGS})
+elif [[ "$#" -ne 0 && "${INPUT_REMARK_FLAGS}" == "" ]]; then
+  remark_args+=($*)
+fi
+
+# Run remark-lint with reviewdog
+remark_exit_val="0"
+reviewdog_exit_val="0"
+echo "[action-remark-lint] Checking markdown code using the remark-lint linter and reviewdog..."
+remark_lint_output="$(remark --frail --quiet --use=remark-preset-lint-recommended ${remark_args[*]} 2>&1)" ||
+  remark_exit_val="$?"
+
+# Input remark formatter output to reviewdog
+echo "${remark_lint_output}" |
   sed 's/\x1b\[[0-9;]*m//g' | # Removes ansi codes see https://github.com/reviewdog/errorformat/issues/51
   reviewdog -f=remark-lint \
-  -name="${INPUT_TOOL_NAME}" \
-  -reporter="${INPUT_REPORTER:-github-pr-check}"\
-  -filter-mode="${INPUT_FILTER_MODE}" \
-  -fail-on-error="${INPUT_FAIL_ON_ERROR}" \
-  -level="${INPUT_LEVEL}" \
-  -tee \
-  ${INPUT_REVIEWDOG_FLAGS}
\ No newline at end of file
+    -name="${INPUT_TOOL_NAME}" \
+    -reporter="${INPUT_REPORTER}" \
+    -filter-mode="${INPUT_FILTER_MODE}" \
+    -fail-on-error="${INPUT_FAIL_ON_ERROR,,}" \
```

Comment:
What does `,,` do? typo?

Reply:
@haya14busa It's bash 4.0 syntax for making strings lowercase (see this StackOverflow post). I will add a comment such that it doesn't confuse other developers.